--- src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs.orig	2012-06-29 19:27:37.131882982 +0200
+++ src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs	2012-06-29 19:28:24.584286423 +0200
@@ -195,9 +195,6 @@
 			if (!CheckBug77135 ())
 				return 1;
 			
-			if (!CheckQtCurve ())
-				return 1;
-
 			CheckFileWatcher ();
 			
 			Exception error = null;
@@ -356,19 +353,6 @@
 			return false;
 		}
 		
-		bool CheckQtCurve ()
-		{
-			if (Gtk.Settings.Default.ThemeName == "QtCurve") {
-				string msg = "QtCurve theme not supported";
-				string desc = "Your system is using the QtCurve GTK+ theme. This theme is known to cause stability issues in MonoDevelop. Please select another theme in the GTK+ Theme Selector.\n\nIf you click on Proceed, MonoDevelop will switch to the default GTK+ theme.";
-				AlertButton res = MessageService.GenericAlert (Gtk.Stock.DialogWarning, msg, desc, AlertButton.Cancel, AlertButton.Proceed);
-				if (res == AlertButton.Cancel)
-					return false;
-				Gtk.Settings.Default.ThemeName = "Gilouche";
-			}
-			return true;
-		}
-		
 		void CheckFileWatcher ()
 		{
 			string watchesFile = "/proc/sys/fs/inotify/max_user_watches";
