--- src/corelib/animation/qpropertyanimation.cpp
+++ src/corelib/animation/qpropertyanimation.cpp
@@ -133,13 +133,7 @@
         return;
     }
 
-    if (newValue.userType() == propertyType) {
-        //no conversion is needed, we directly call the QMetaObject::metacall
-        void *data = const_cast<void*>(newValue.constData());
-        QMetaObject::metacall(targetValue, QMetaObject::WriteProperty, propertyIndex, &data);
-    } else {
-        targetValue->setProperty(propertyName.constData(), newValue);
-    }
+    targetValue->setProperty(propertyName.constData(), newValue);
 }
 
 /*!
