--- a/strippath.c	2012-01-21 14:55:21.000000000 +0100
+++ b/strippath.c	2012-02-03 23:47:25.258089908 +0100
@@ -106,8 +106,10 @@
 	e->f_name_size -= len;
 	e->f_name[e->f_name_size] = '\0';
 	
-	if (S_ISLNK(e->f_mode) || e->f_lnk == 1)
+	if (S_ISLNK(e->f_mode) || e->f_lnk == 1) {
 		e->f_size -= len;
+		e->f_target -= len;
+	}
 
 	/* hardlinks need to get the same treatment */
 	if (e->f_lnk == 1) {
